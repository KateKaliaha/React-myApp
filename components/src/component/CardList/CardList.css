.card-list {
  width: 90%;
  display: flex;
  flex-wrap: wrap;
  justify-content: start;
  gap: 50px;
  margin-top: 30px;
}
