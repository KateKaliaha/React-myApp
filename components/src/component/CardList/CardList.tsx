import { Card } from 'component/Card/Card';
import { ICardApi } from 'data/interfaces';
import React from 'react';
import { useNavigate } from 'react-router-dom';
import { useAppDispatch, useAppSelector } from '../../hook';
import { setNewDisplayStyle } from 'store/mainPageSlice';

export function CardList(): JSX.Element {
  const data = useAppSelector((state) => state.movie.movies);
  const dispatch = useAppDispatch();

  const navigate = useNavigate();

  function openMovieCard(event: React.MouseEvent): void {
    const id = +((event.target as HTMLElement).closest('.card')?.getAttribute('id') as string);

    if (id) {
      navigate(`${id}`);
      dispatch(setNewDisplayStyle('none'));
    }
  }

  return (
    <div className="card-list" onClick={openMovieCard}>
      {data.map((card: ICardApi, i) => (
        <Card movie={card} key={card.id + '' + i} />
      ))}
    </div>
  );
}
