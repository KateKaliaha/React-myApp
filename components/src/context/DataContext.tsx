import { ICardApi, IFormCard } from 'data/interfaces';
import { Action, ReducerContextValue, ReducerState } from 'data/types';
import React, { createContext, ReactNode, Reducer, useReducer } from 'react';

const initialState = {
  display: 'flex',
  data: [],
  value: null,
  inputValue: undefined,
  page: 1,
  totalPages: undefined,
  totalResults: undefined,
  countMovieOnPage: 20,
  card: undefined,
  isFirstLoad: false,
  sort: 'popularity.desc',
  cardForm: [],
};

const reducer: Reducer<ReducerState, Action> = (state, action): ReducerState => {
  switch (action.type) {
    case 'newDisplayStyle':
      return { ...state, display: action.payload as string };
    case 'newData':
      return { ...state, data: action.payload as ICardApi[] };
    case 'newValue':
      return { ...state, value: action.payload as string | null };
    case 'newInputValue':
      return { ...state, inputValue: action.payload as string | undefined };
    case 'newPage':
      return { ...state, page: action.payload as number };
    case 'newTotalPages':
      return { ...state, totalPages: action.payload as number };
    case 'newTotalResults':
      return { ...state, totalResults: action.payload as number };
    case 'newCountMovieOnPage':
      return { ...state, countMovieOnPage: action.payload as number };
    case 'newCard':
      return { ...state, card: action.payload as ICardApi };
    case 'newIsFirstLoad':
      return { ...state, isFirstLoad: action.payload as boolean };
    case 'newSort':
      return { ...state, sort: action.payload as string };
    case 'newFormCard':
      return { ...state, cardForm: action.payload as IFormCard[] };
    default:
      throw new Error();
  }
};

const DataContext = createContext<ReducerContextValue>({
  state: initialState,
  dispatch: () => {},
});

export const DataProvider: React.FC<{ children: ReactNode }> = ({ children }) => {
  const [state, dispatch] = useReducer(reducer, initialState);

  return <DataContext.Provider value={{ state, dispatch }}>{children}</DataContext.Provider>;
};

export default DataContext;
