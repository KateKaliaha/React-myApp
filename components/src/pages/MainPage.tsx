import React from 'react';
import { Search } from 'component/Search/Search';

function MainPage() {
  return (
    <div className="main-page">
      <Search />
    </div>
  );
}
export { MainPage };
