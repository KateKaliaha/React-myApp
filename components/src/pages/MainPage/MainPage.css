.main-page {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  margin: 20px;
}

.pages-wrapper {
  display: flex;
  flex-wrap: wrap;
  margin: 0 auto;
  width: 90vw;
  font-size: 1.3rem;
  gap: 2px;
  margin-bottom: 10px;
}

.page-btn {
  cursor: pointer;
  border: 2px solid #3f4348;
  border-radius: 8px;
  width: 40px;
  text-align: center;
  background-color: #e9e9e9;
}

.page-btn.active-page-btn {
  background-color: rgb(202, 75, 157);
}

.select-wrapper {
  font-size: 2rem;
  display: flex;
  margin: 0 auto;
  width: 90vw;
  gap: 10px;
  margin-bottom: 20px;
  text-align: center;
  color: #3f4348;
}

.select-sort {
  height: 30px;
}
