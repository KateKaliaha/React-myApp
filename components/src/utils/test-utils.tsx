import { combineReducers, configureStore } from '@reduxjs/toolkit';
import type { PreloadedState } from '@reduxjs/toolkit';
import MovieReducer from '../store/movieSlice';
import PageReducer from '../store/pageComponentSlice';
import MainPageReducer from '../store/mainPageSlice';
import FormCardReducer from '../store/formCardSlice';

import React, { PropsWithChildren } from 'react';
import { render } from '@testing-library/react';
import type { RenderOptions } from '@testing-library/react';

import { Provider } from 'react-redux';

import type AppStore from '../store/index';
// Create the root reducer separately so we can extract the RootState type
const rootReducer = combineReducers({
  movie: MovieReducer,
  pageComponent: PageReducer,
  mainPage: MainPageReducer,
  formCard: FormCardReducer,
});

export const setupStore = (preloadedState?: PreloadedState<RootState>) => {
  return configureStore({
    reducer: rootReducer,
    preloadedState,
  });
};

export type RootState = ReturnType<typeof rootReducer>;
export type AppStore = ReturnType<typeof setupStore>;
export type AppDispatch = AppStore['dispatch'];

// This type interface extends the default options for render from RTL, as well
// as allows the user to specify other things such as initialState, store.
interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>;
  store?: AppStore;
}

export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = setupStore(preloadedState),
    ...renderOptions
  }: ExtendedRenderOptions = {}
) {
  function Wrapper({ children }: PropsWithChildren<Record<string, unknown>>): JSX.Element {
    return <Provider store={store}>{children}</Provider>;
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}
