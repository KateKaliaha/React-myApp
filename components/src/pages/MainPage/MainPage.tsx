import React, { Component } from 'react';
import { Search } from 'component/Search/Search';
import { CardList } from 'component/CardList/CardList';
import { Message } from 'component/Message/Message';
import { ICardApi, IResponse } from 'data/interfaces';
import { CardsProps, CardsState } from 'data/types';
import { ModalWindow } from 'component/ModalWindow/ModalWindow';

const PATH = 'https://api.themoviedb.org/3/';
const API_KEY = 'api_key=9c5e0f16891cead9f73032e139a5c245';
const LANG_API_RESPONSE = 'ru-Ru';

class MainPage extends Component<CardsProps, CardsState> {
  page: number;

  constructor(props: CardsProps) {
    super(props);
    this.page = 1;
    this.state = {
      data: [],
      modalActive: false,
      card: undefined,
    };
    this.getSearchCardList = this.getSearchCardList.bind(this);
    this.openModalWindow = this.openModalWindow.bind(this);
    this.closeModalWindow = this.closeModalWindow.bind(this);
  }

  async componentDidMount(): Promise<void> {
    const localValue = localStorage.getItem('value') as string;
    if (localValue) {
      await this.getApiData('search', `query=${localValue}&page=${this.page}`);
    } else {
      await this.getApiData('discover', `page=${this.page}`);
    }
  }

  async getApiData(param: string, value: string) {
    await fetch(`${PATH}${param}/movie?${API_KEY}&language=${LANG_API_RESPONSE}&${value}`)
      .then((resp) => {
        if (resp.ok) {
          return resp.json() as Promise<IResponse>;
        }
        throw resp;
      })
      .then((data) => {
        if (param === 'search') {
          const valueForFilter = value.split('&')[0].split('=')[1];
          const filterData = data.results.filter((movie) =>
            movie.title.toLowerCase().includes(valueForFilter.toLowerCase())
          );
          this.setState({ data: [...filterData] });
        } else {
          this.setState({ data: [...data.results] });
        }
      })
      .catch((error) => console.error(error));
  }

  async getSearchCardList(value: string) {
    await this.getApiData('search', `query=${value}&page=${this.page}`);
  }

  async openModalWindow(event: React.MouseEvent): Promise<void> {
    const id = +((event.target as HTMLElement).closest('.card')?.getAttribute('id') as string);
    if (id) {
      const card = this.state.data.find((el) => el.id === id);

      await this.setState({ modalActive: true, card: card });
    }
  }

  async closeModalWindow(): Promise<void> {
    await this.setState({ modalActive: false });
  }

  render() {
    return (
      <div className="main-page">
        <Search getSearchCardList={this.getSearchCardList} />
        {this.state.data.length > 0 && (
          <CardList data={this.state.data} openModalWindow={this.openModalWindow} />
        )}
        {this.state.data.length === 0 && <Message />}
        {this.state.modalActive && (
          <ModalWindow
            active={this.state.modalActive}
            movie={this.state.card as ICardApi}
            closeModalWindow={this.closeModalWindow}
          />
        )}
      </div>
    );
  }
}

export { MainPage };
