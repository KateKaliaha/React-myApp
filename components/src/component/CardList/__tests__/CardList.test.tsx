import { cleanup, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import DataContext from 'context/DataContext';
import { state, dispatch } from 'pages/MainPage/MainPage.test';
import React from 'react';
import { act } from 'react-dom/test-utils';
import { BrowserRouter } from 'react-router-dom';
import { CardList } from '../CardList';

describe('CardList', () => {
  afterEach(cleanup);

  it('snapshot CardList container', async () => {
    const { container } = render(
      <BrowserRouter>
        <DataContext.Provider value={{ state, dispatch }}>
          <CardList />
        </DataContext.Provider>
      </BrowserRouter>
    );

    expect(container).toMatchSnapshot();
  });

  it('click on card', async () => {
    render(
      <BrowserRouter>
        <DataContext.Provider value={{ state, dispatch }}>
          <CardList />
        </DataContext.Provider>
      </BrowserRouter>
    );
    const card = screen.queryAllByTestId('card');
    await act(async () => await userEvent.click(card[0]));
  });
});
