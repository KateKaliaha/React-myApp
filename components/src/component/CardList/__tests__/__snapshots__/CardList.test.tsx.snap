// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList snapshot CardList container 1`] = `
<div>
  <div
    class="card-list"
  >
    <div
      class="card"
      data-testid="card"
      id="1021735"
    >
      <img
        class="card-img"
        data-testid="img-card"
        src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAALgAAAETCAMAAACyQTxsAAAAV1BMVEX///+em5ubmJignZ2Zlpain5/t7OzS0dH29vb8/PzIxsbDwcHBv7/Y1tbo5+fMysqppqa4tbWyr6/f3t7c29vy8vK0srLPzs6tqqrk4+O7ubmmpKOTj4/0wm4IAAAHiUlEQVR4nO2b69aiOgyGTVrOKIiogHP/17lLD7TwgePMkq2z1vv8Ag3wimnaJHA4AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAD+jDSOM78Xx1e/c0qq6n7Jfh7zFeRSNn5PysFtVsSGMv2Ert9SUcTHaY+oNBtpxIKj5lZLIr58SNtTlPCIp3vqhKckuIn1Vi6FbD8k7hlKuBCd23PCa8F399mJI3ldOfLDVCTOghO7Z4UfmQpvcmFx+4Cy36DGYCaEcxYr/CE4tGnEF97yiuShl8JGFiP8Kt0YNbRM95VDP4sSnh0Gss5ihLfMs+GYsjh/QttTtPADWWcxwpX7nGZGJB6f0PYUI1w5i44sRnhBPPfphxCf0PYUI9w5y78n3EYWIzwnjmdG3+sqzlmM8ONicGbfGMidcOMs6+Hw8rXhUCOETIVRrGb8cDX7rROQ1ThOQ5ERrmacwZt865TvhCtnEVb4oROcO4uvXWRNwpWzOOGp8MtaFvK4cfAnCYX3MpoSiZpJJxKsEolvXI6r1O2XH4dlkLol4rtTN5UsB3vLZDm/99+aLAMAAAAA/Fuks5XWE7PT3ulFQcLw6Epbo1d5vK9VpdLXZuPSrGUH+/VVihm2GJrmkVRWstl1qV5QZNsiKil4pD+EZ154OTYgtKG8ZUa4OTCyp5DaLLFmKtl47HjbC6K7oVD3LHoivGNBVZxlcRIRRaPyVB+XDIKKRG/qE0rBZZ9m11YdsCgzvle4nLY7U8RfF34maqbMmUXtT6Hsfb6RSBG5+5yw4N0ypCJIK6+sq5urwn2ZfORMvpsys8+Ubp8WJUy71S5C4bbZsCq8m928jARNO6F9Pm/DdftVi2bC603hKoAM4WFKYO+2Q3s7TBw9U37Yh1B4tu0qyz7EKVAU2F95Xlg8sG/avZlQeElUHUxbLXHcSQtfFpezwOUD4RemJDQbfWU34RxfR+K2UfEu00JcaNdBWgsfaBEfhI8rgXC1Oe80n4XcqYShJiCnUZA8matHAcIIn5doD4fIl/MD4cmyRf5/CJeiSJdCRr9fv+P02h2/7SicT7FmErY2OKufPn5ese+XPr7o5r6RWTj8IWQSvhx1fdCHCOzTRdTcsc/yovBsEdcGkvGafTcfC+qr+T/wPl4UrkJl6L1qPqpX7dvZEwpqCPNeddFXhSsnoGn2ziIh+3V75dR+plJLmurtii2vCj+0cpJ0Ujey3LBX/4UTmzZEwSLyzbwsfFyw8iM/tpVaZ8thy35cDJAojm1yVovaer8Cesk/hUsZpm7snDYeOyg6sxHtpr064iat2X5+omjzYin8lOd+LZoVuVfZF10kptx01X4krho1sQ5H9CsAAAAAAL6aS+6WekU+e+bumucmo2nz/DKz13Vw9+2MNvxUmRiqo8tFs9mpFFUe8ke5aKQW+WarYRnWSXL+ZVamRBRULQtjFUu58mjh2HGYdmLfhpAPI12t5OcFTyJfIGP+k5riiVXOZe7CotwgbCWwHy18VpCbMlC8rJaYM4SmMYtHPRJJEqYAkC4rtZGIas+fPLtd8nCmxkkN0sHe1WLPXDTkT/lM+I3zzj+AqEzs03BZZdsVK8L/tsbCfGzZVtpzCkruZ+tBGfPlGBTbnghXjtDfaSo+hCaDqeavCG8Of8Vx9Fhpz3aVNGWEmSuGJCpzTqWvUT0RPopWpu7z0OTCuuj/PuEddWMJyg7PzvcO1C8y46keHanxw/OJ8MfY3emmku1C+Omdwq9yLIdN7tz6okJNnb36+F3LU5dkW/hJl2V9YSI0sY73NuG58UhyRcCp66F+kRlYBenSqi9UbgsvtbhserHDD85ryfKyIfzvBqcw4SJ35cqSePaLphd9SnIxflu4HRWDM41VcHQd6sj8YWvhMOgXvKy7tx4Qu9JU7Irw9rnwyXlO7MbtpnDXk+2d6SjcNtZFlG8JF37+eb01dHZjrnO19tr8Bb1z1IbsUHu4Juam8M7NApEd4moCGo662VU82DbMfwqv79XEq7ozKTrdcE9uwrp0wtpFzlZvKkVjLBo3v24Jv0pxm0zN/xiYVKzvyJsGZ0KTD7pJU003Rx3EzSWr0KJ5KjwPTW9L4WOPMH2b8FpEw1kz1HYxoYZWp4N4Zs/7CCyuz4QLb/oQ8oeJeTnrPcLDE59cIFTenapp6Ww/9S/LXuw1N4T3QZu5NcNzN+EFBcvYyPXwBCfjmkNvDxR0PdgMzw3h57CVxvrFq91chcLeU+XeF6y4ObIJC9nsJcHCvEKwLjybvZZimkOhSc5a4VuEt4Ef6OdSpkBS2+r7vLUam/fyJuFimHr7l7F9HCREJy1zMlHhkNzqUIUed1SrhdeJ5/7S8yDNvGGi/ksXuoV1oXreSX3oAZyTHXliyl1kuWy6RqNOb6LSHDrZu+InnNthMQHxrIuxhfrxs75pS3b/QpFZP6gLz57XUNFTza+5sHc8mKpzZRp2BA93IXJv8uiGo7tocNR4uWgG7fe4FgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA+CD/AbS6UfORUBZtAAAAAElFTkSuQmCC"
      />
      <h2
        class="card-header"
      >
        Продаётся дом
      </h2>
      <div
        class="popular"
      >
        <div
          class="rate-container"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 576 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
            />
          </svg>
          <p>
            0
          </p>
        </div>
        <div
          class="like-container"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
            />
          </svg>
          <p>
            0
          </p>
        </div>
      </div>
    </div>
    <div
      class="card"
      data-testid="card"
      id="79580"
    >
      <img
        class="card-img"
        data-testid="img-card"
        src="https://image.tmdb.org/t/p/w500/iSnRfGxEeX9O2rTHcWn38qR5D1F.jpg"
      />
      <h2
        class="card-header"
      >
        Дом
      </h2>
      <div
        class="popular"
      >
        <div
          class="rate-container"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 576 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
            />
          </svg>
          <p>
            6.3
          </p>
        </div>
        <div
          class="like-container"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
            />
          </svg>
          <p>
            26
          </p>
        </div>
      </div>
    </div>
    <div
      class="card"
      data-testid="card"
      id="70070"
    >
      <img
        class="card-img"
        data-testid="img-card"
        src="https://image.tmdb.org/t/p/w500/s2KKdUusmKoCNIR0o5Byus4EcJO.jpg"
      />
      <h2
        class="card-header"
      >
        Дом Солнца
      </h2>
      <div
        class="popular"
      >
        <div
          class="rate-container"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 576 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
            />
          </svg>
          <p>
            5.1
          </p>
        </div>
        <div
          class="like-container"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
            />
          </svg>
          <p>
            16
          </p>
        </div>
      </div>
    </div>
    <div
      class="card"
      data-testid="card"
      id="72996"
    >
      <img
        class="card-img"
        data-testid="img-card"
        src="https://image.tmdb.org/t/p/w500/4jpGDmOgHRbpCe36SIYZKi3TXiy.jpg"
      />
      <h2
        class="card-header"
      >
        Дом дураков
      </h2>
      <div
        class="popular"
      >
        <div
          class="rate-container"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 576 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
            />
          </svg>
          <p>
            6
          </p>
        </div>
        <div
          class="like-container"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
            />
          </svg>
          <p>
            30
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
